implant/sliver/ps/ps_darwin.go

```diff
@@ -227,12 +227,9 @@ func getArgvFromPid(pid int) ([]string, error) {
 		errStr := unix.ErrnoName(errno)
 		return []string{""}, fmt.Errorf("%s", errStr)
 	}
-	buffer := bytes.NewBuffer(processArgs)
-	numberOfArgs, err := binary.ReadUvarint(buffer)
-	if err != nil {
-		return []string{""}, err
-	}
-	buffer.Next(3)                         // skip  sizeof(int32), the number of args
+	buffer := bytes.NewBuffer(processArgs[0:size])
+	numberOfArgsBytes := buffer.Read(4)
```

Comment:
Please fix this call before we can test it. `buffer.Read()` takes a `[]byte` as argument, not an int:

```go
package bytes // import "bytes"

func (b *Buffer) Read(p []byte) (n int, err error)
    Read reads the next len(p) bytes from the buffer or until the buffer is
    drained. The return value n is the number of bytes read. If the buffer has
    no data to return, err is io.EOF (unless len(p) is zero); otherwise it is
    nil.

func (r *Reader) Read(b []byte) (n int, err error)
    Read implements the io.Reader interface.
```

Reply:
So sorry about the rookie mistake.  It's been fixed. I amended my commit and did a forced push to my branch to keep the commit log clean.

However, I tried to test `sliver-server` but got the follow error. I did run `go generate` on `implant/` which did not change any files. Also removed `~/.sliver` but the issue remains. Any idea where to look at?

```sh
Sliver  Copyright (C) 2022  Bishop Fox
This program comes with ABSOLUTELY NO WARRANTY; for details type 'licenses'.
This is free software, and you are welcome to redistribute it
under certain conditions; type 'licenses' for details.

Unpacking assets ...
panic: runtime error: index out of range [1] with length 1

goroutine 101 [running]:
github.com/bishopfox/sliver/server/rpc.(*Server).GetVersion(0xd5e9630?, {0x2a3a500?, 0x2a914ac?}, 0xf?)
	github.com/bishopfox/sliver/server/rpc/rpc.go:90 +0x13d
github.com/bishopfox/sliver/protobuf/rpcpb._SliverRPC_GetVersion_Handler.func1({0xc023ca8?, 0xc0007ca8a0?}, {0x27c2f80?, 0xc0007ca300?})
	github.com/bishopfox/sliver/protobuf/rpcpb/services_grpc.pb.go:3006 +0xce
github.com/bishopfox/sliver/server/transport.initMiddleware.PayloadUnaryServerInterceptor.func16({0xc023ca8, 0xc0007ca8a0}, {0x27c2f80, 0xc0007ca300}, 0x0?, 0xc00341c348)
	github.com/grpc-ecosystem/go-grpc-middleware@v1.4.0/logging/logrus/payload_interceptors.go:29 +0x179
google.golang.org/grpc.getChainUnaryHandler.func1({0xc023ca8, 0xc0007ca8a0}, {0x27c2f80, 0xc0007ca300})
	google.golang.org/grpc@v1.64.0/server.go:1196 +0xb2
github.com/grpc-ecosystem/go-grpc-middleware/logging/logrus.UnaryServerInterceptor.func1({0xc023ca8, 0xc0007ca7b0}, {0x27c2f80, 0xc0007ca300}, 0xc00076c100, 0xc000758700)
	github.com/grpc-ecosystem/go-grpc-middleware@v1.4.0/logging/logrus/server_interceptors.go:31 +0xfe
google.golang.org/grpc.getChainUnaryHandler.func1({0xc023ca8, 0xc0007ca7b0}, {0x27c2f80, 0xc0007ca300})
	google.golang.org/grpc@v1.64.0/server.go:1196 +0xb2
github.com/grpc-ecosystem/go-grpc-middleware/tags.UnaryServerInterceptor.func1({0xc023ca8?, 0xc0007ca360?}, {0x27c2f80, 0xc0007ca300}, 0xc00076c100, 0xc0007586c0)
	github.com/grpc-ecosystem/go-grpc-middleware@v1.4.0/tags/interceptors.go:23 +0x9c
google.golang.org/grpc.getChainUnaryHandler.func1({0xc023ca8, 0xc0007ca360}, {0x27c2f80, 0xc0007ca300})
	google.golang.org/grpc@v1.64.0/server.go:1196 +0xb2
github.com/bishopfox/sliver/server/transport.initMiddleware.auditLogUnaryServerInterceptor.func14({0xc023ca8, 0xc0007ca360}, {0x27c2f80, 0xc0007ca300}, 0xc00076c100, 0xc000758600)
	github.com/bishopfox/sliver/server/transport/middleware.go:317 +0x3e8
google.golang.org/grpc.getChainUnaryHandler.func1({0xc023ca8, 0xc0007ca360}, {0x27c2f80, 0xc0007ca300})
	google.golang.org/grpc@v1.64.0/server.go:1196 +0xb2
github.com/bishopfox/sliver/server/transport.initMiddleware.UnaryServerInterceptor.func13({0xc023ca8?, 0xc0007ca2d0?}, {0x27c2f80, 0xc0007ca300}, 0xc00076c100?, 0xc0007585c0)
	github.com/grpc-ecosystem/go-grpc-middleware@v1.4.0/auth/auth.go:47 +0x91
google.golang.org/grpc.NewServer.chainUnaryServerInterceptors.chainUnaryInterceptors.func1({0xc023ca8, 0xc0007ca2d0}, {0x27c2f80, 0xc0007ca300}, 0xc00076c100, 0x80?)
	google.golang.org/grpc@v1.64.0/server.go:1187 +0x85
github.com/bishopfox/sliver/protobuf/rpcpb._SliverRPC_GetVersion_Handler({0x2a3a500, 0xd6c0900}, {0xc023ca8, 0xc0007ca2d0}, 0xc0005ae100, 0xc00023a460)
	github.com/bishopfox/sliver/protobuf/rpcpb/services_grpc.pb.go:3008 +0x143
google.golang.org/grpc.(*Server).processUnaryRPC(0xc000386e00, {0xc023ca8, 0xc0007ca240}, {0xc0347e0, 0xc000002480}, 0xc00343c7e0, 0xc0004e6ff0, 0xd64a0c0, 0x0)
	google.golang.org/grpc@v1.64.0/server.go:1379 +0xdf8
google.golang.org/grpc.(*Server).handleStream(0xc000386e00, {0xc0347e0, 0xc000002480}, 0xc00343c7e0)
	google.golang.org/grpc@v1.64.0/server.go:1790 +0xe8b
google.golang.org/grpc.(*Server).serveStreams.func2.1()
	google.golang.org/grpc@v1.64.0/server.go:1029 +0x8b
created by google.golang.org/grpc.(*Server).serveStreams.func2 in goroutine 78
	google.golang.org/grpc@v1.64.0/server.go:1040 +0x125

```